core/src/main/java/me/aurium/beetle/api/task/sync/SyncFuture.java

```diff
@@ -0,0 +1,68 @@
+package me.aurium.beetle.api.task.sync;
+
+import me.aurium.beetle.api.task.TaskFuture;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * i want to stop existing
+ *
+ * A safer to use Future that permits get and join based on an internal behavior.
+ *
+ * TODO eventually some sort of resync to main thread functionality
+ *
+ * @param <T>
+ */
+public class SyncFuture<T> extends TaskFuture<T> {
+
+
+    private final TaskCoordinator coordinator;
+
+    public SyncFuture(TaskCoordinator coordinator) {
+        this.coordinator = coordinator;
+    }
+
+
+    @Override
+    public T get(long amount, TimeUnit unit) throws ExecutionException, InterruptedException, TimeoutException {
+        if (coordinator.isMainThread()) {
```

Comment:
These method implementations have improved, but there's still an issue. A SyncFuture can return a null result, so future.getNow(null) is ambiguous

Reply:
im going to cease to exist